homeassistant/components/knx/sensor.py

```diff
@@ -93,6 +98,11 @@ def unit_of_measurement(self):
         """Return the unit this state is expressed in."""
         return self.device.unit_of_measurement()
 
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return self.device.ha_device_class()
```

Comment:
Is xknx aware of home assistant sensor device classes?

Reply:
Yes, xknx sensor classes are. The device class is returned as a string - or `None` if there is no   fitting HA device class.
https://github.com/XKNX/xknx/blob/cc10a82d11b4a60bd9765a7b410ab05edd5da41f/xknx/devices/remote_value_sensor.py#L121

Reply:
Ok. Coupling the two packages this tightly is fragile. If the home assistant API would change, something could break until the xknx library is updated. I would recommend to have a dict in this module that maps between the xknx sensor class and the home assistant device class.

But since the xknx library seems to be built for home assistant I don't think it's strictly necessary.

Reply:
Xknx works on its own but is built with HA as a main target.
With almost 50 different sensor classes its nice to have everything in one place (range, unit of measurement, ha_device_class, ...) instead of spreading these attributes out to different modules.

We could import the DEVICE_CLASS_* constants in this module and map against our ha_device_class attribute. To me this would feel kind of redundant.